api/api_test.go

```diff
@@ -18,11 +19,24 @@ func TestSetup(t *testing.T) {
 		ctx := context.Background()
 
 		mockMongoDB := &mock.DataStoreMock{}
-		cacheAPI := api.Setup(ctx, router, mockMongoDB)
 
-		Convey("When created the following routes should have been added", func() {
-			So(hasRoute(cacheAPI.Router, "/v1/cache-times/{id}", "GET"), ShouldBeTrue)
-			So(hasRoute(cacheAPI.Router, "/v1/cache-times/{id}", "PUT"), ShouldBeTrue)
+		Convey("When created in publishing mode", func() {
+			cfg := &config.Config{IsPublishing: true}
+			cacheAPI := api.Setup(ctx, cfg.IsPublishing, router, mockMongoDB)
+
+			Convey("Then the following routes should have been added", func() {
+				So(hasRoute(cacheAPI.Router, "/v1/cache-times/{id}", "GET"), ShouldBeTrue)
+				So(hasRoute(cacheAPI.Router, "/v1/cache-times/{id}", "PUT"), ShouldBeTrue)
+			})
+		})
+		Convey("When created in web mode", func() {
+			cfg := &config.Config{IsPublishing: false}
+			cacheAPI := api.Setup(ctx, cfg.IsPublishing, router, mockMongoDB)
+
+			Convey("Then the get endpoint should have been added", func() {
```

Comment:
I would call this "Then the PUT endpoint should not have been added", since this is the behaviour we want.

Similarly, instead of "Then the following routes should have been added" you could say "Then all the routes should be available". By doing it this way, you are clearly stating what we would expect in both scenarios. The test cases read as the actual requirements.